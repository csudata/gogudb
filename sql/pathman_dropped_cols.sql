\set VERBOSITY terse


SET search_path = 'public','_gogu';
CREATE EXTENSION gogudb;
CREATE SCHEMA dropped_cols;


/*
 * we should be able to manage tables with dropped columns
 */

create table test_range(a int, b int, key int not null);

alter table test_range drop column a;
select create_range_partitions('test_range', 'key', 1, 10, 2);

alter table test_range drop column b;
select prepend_range_partition('test_range');

select * from gogudb_partition_list order by parent, partition;
select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_1_test_range_check';
select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_3_test_range_check';

drop table test_range cascade;


create table test_hash(a int, b int, key int not null);

alter table test_hash drop column a;
select create_hash_partitions('test_hash', 'key', 3);

alter table test_hash drop column b;
create table test_dummy (like test_hash);
select replace_hash_partition('gogudb_partition_table._public_2_test_hash', 'test_dummy', true);

select * from gogudb_partition_list order by parent, partition;
select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_1_test_hash_check';
select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb_test_dummy_check';
drop table test_hash cascade;


DROP SCHEMA dropped_cols CASCADE;
DROP EXTENSION gogudb CASCADE;

