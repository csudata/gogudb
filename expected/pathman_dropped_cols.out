\set VERBOSITY terse
SET search_path = 'public','_gogu';
CREATE EXTENSION gogudb;
CREATE SCHEMA dropped_cols;
/*
 * we should be able to manage tables with dropped columns
 */
create table test_range(a int, b int, key int not null);
alter table test_range drop column a;
select create_range_partitions('test_range', 'key', 1, 10, 2);
 create_range_partitions 
-------------------------
                       2
(1 row)

alter table test_range drop column b;
select prepend_range_partition('test_range');
           prepend_range_partition           
---------------------------------------------
 gogudb_partition_table._public_3_test_range
(1 row)

select * from gogudb_partition_list order by parent, partition;
   parent   |                  partition                  | parttype | expr | range_min | range_max 
------------+---------------------------------------------+----------+------+-----------+-----------
 test_range | gogudb_partition_table._public_1_test_range |        2 | key  | 1         | 11
 test_range | gogudb_partition_table._public_2_test_range |        2 | key  | 11        | 21
 test_range | gogudb_partition_table._public_3_test_range |        2 | key  | -9        | 1
(3 rows)

select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_1_test_range_check';
     pg_get_constraintdef      
-------------------------------
 CHECK (key >= 1 AND key < 11)
(1 row)

select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_3_test_range_check';
           pg_get_constraintdef           
------------------------------------------
 CHECK (key >= '-9'::integer AND key < 1)
(1 row)

drop table test_range cascade;
NOTICE:  drop cascades to 4 other objects
create table test_hash(a int, b int, key int not null);
alter table test_hash drop column a;
select create_hash_partitions('test_hash', 'key', 3);
 create_hash_partitions 
------------------------
                      3
(1 row)

alter table test_hash drop column b;
create table test_dummy (like test_hash);
select replace_hash_partition('gogudb_partition_table._public_2_test_hash', 'test_dummy', true);
 replace_hash_partition 
------------------------
 test_dummy
(1 row)

select * from gogudb_partition_list order by parent, partition;
  parent   |                 partition                  | parttype | expr | range_min | range_max 
-----------+--------------------------------------------+----------+------+-----------+-----------
 test_hash | gogudb_partition_table._public_0_test_hash |        1 | key  |           | 
 test_hash | gogudb_partition_table._public_1_test_hash |        1 | key  |           | 
 test_hash | test_dummy                                 |        1 | key  |           | 
(3 rows)

select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb__public_1_test_hash_check';
                                        pg_get_constraintdef                                        
----------------------------------------------------------------------------------------------------
 CHECK (get_hash_part_idx(hashint4(key), 128) >= 43 AND get_hash_part_idx(hashint4(key), 128) < 86)
(1 row)

select pg_get_constraintdef(oid, true) from pg_constraint where conname = 'gogudb_test_dummy_check';
                                        pg_get_constraintdef                                         
-----------------------------------------------------------------------------------------------------
 CHECK (get_hash_part_idx(hashint4(key), 128) >= 86 AND get_hash_part_idx(hashint4(key), 128) < 128)
(1 row)

drop table test_hash cascade;
NOTICE:  drop cascades to 3 other objects
DROP SCHEMA dropped_cols CASCADE;
DROP EXTENSION gogudb CASCADE;
NOTICE:  drop cascades to table gogudb_partition_table._public_2_test_hash
